### PR TITLE
#63365: ProjectPlanningTime : Manage project units and improve UX

### PR DESCRIPTION
**Pull request by cce-axelor**

### CODE REVIEW

File: axelor-human-resource/src/main/java/com/axelor/apps/hr/service/timesheet/TimesheetServiceImpl.java [hunk 1]
```diff
@@ -1205,18 +1205,18 @@ protected TimesheetLine createTimeSheetLineFromPPT(
...
+            timesheet, projectPlanningTime.getPlannedTime(), false));
     timesheetLine.setTimesheet(timesheet);
     timesheetLine.setEmployee(timesheet.getEmployee());
     timesheetLine.setProduct(projectPlanningTime.getProduct());
     if (project.getIsShowTimeSpent()) {
       timesheetLine.setProjectTask(projectPlanningTime.getProjectTask());
       timesheetLine.setProject(projectPlanningTime.getProject());
     }
-    timesheetLine.setDate(projectPlanningTime.getDate());
+    timesheetLine.setDate(projectPlanningTime.getStartDateTime().toLocalDate());
```
Comment (ale-axelor) [inline axelor-human-resource/src/main/java/com/axelor/apps/hr/service/timesheet/TimesheetServiceImpl.java:1219]: can you check if `startDateTime` can be null here ? With the previous code it did not cause any issue if the field was null, but now the call to `.toLocalDate()` will throw a NPE

**Review by ale-axelor — CHANGES_REQUESTED** [review]

File: axelor-business-project/src/main/java/com/axelor/apps/businessproject/service/app/AppBusinessProjectServiceImpl.java [hunk 2]
```diff
@@ -51,4 +57,28 @@ public AppBusinessProjectServiceImpl(
...
     return appBusinessProjectRepo.all().fetchOne();
   }
+
+  @Override
+  public Unit getDaysUnit() throws AxelorException {
+    if (Objects.isNull(getAppBusinessProject().getDaysUnit())) {
+      throw new AxelorException(
+          getAppBusinessProject(),
+          TraceBackRepository.CATEGORY_CONFIGURATION_ERROR,
+          I18n.get(BusinessProjectExceptionMessage.PROJECT_CONFIG_DEFAULT_HOURS_PER_DAY_MISSING));
```
Comment (cce-axelor) [inline axelor-business-project/src/main/java/com/axelor/apps/businessproject/service/app/AppBusinessProjectServiceImpl.java:67]: PROJECT_CONFIG_DEFAULT_HOURS_PER_DAY_MISSING message is not adapted

File: axelor-business-project/src/main/java/com/axelor/apps/businessproject/service/app/AppBusinessProjectServiceImpl.java [hunk 3]
```diff
@@ -51,4 +57,28 @@ public AppBusinessProjectServiceImpl(
...
+    return getAppBusinessProject().getDaysUnit();
+  }
+
+  @Override
+  public Unit getHoursUnit() throws AxelorException {
+    if (Objects.isNull(getAppBusinessProject().getHoursUnit())) {
+      throw new AxelorException(
+          getAppBusinessProject(),
+          TraceBackRepository.CATEGORY_CONFIGURATION_ERROR,
+          I18n.get(BusinessProjectExceptionMessage.PROJECT_CONFIG_DEFAULT_HOURS_PER_DAY_MISSING));
```
Comment (cce-axelor) [inline axelor-business-project/src/main/java/com/axelor/apps/businessproject/service/app/AppBusinessProjectServiceImpl.java:79]: Same here

File: axelor-business-project/src/main/java/com/axelor/apps/businessproject/service/app/AppBusinessProjectServiceImpl.java [hunk 4]
```diff
@@ -51,4 +57,28 @@ public AppBusinessProjectServiceImpl(
...
+    }
+
+    return getAppBusinessProject().getHoursUnit();
```
Comment (cce-axelor) [inline axelor-business-project/src/main/java/com/axelor/apps/businessproject/service/app/AppBusinessProjectServiceImpl.java:82]: Store getAppBusinessProject().getDaysUnit() and getAppBusinessProject().getHoursUnit() in a variable

**Pull request merged**